### ISSUE TITLE
Unify Attachment model and profile pictures.

### ISSUE THREAD

**Issue by epicmonkey** [pepyatka/pepyatka-server#186]
e.g. both of them resize images -- could be extracted as a generic layer.

**Comment by yole** [comment]
I intentionally did not reuse the attachment model. A profile picture doesn't have its own identity (it's just a property of the user), so there is no need to have it as a separate model. And the image resizing logic is very different (profile pictures are resized to several different sizes and also cropped to square format).

**Comment by epicmonkey** [comment]
This is right, though we can think about image processing as a library/service that profile pictures and attachment will use. For example, that library will have public interface with a method `hasAttachedFile` and so we can design code in User and Post models like this one:
```
User.hasAttachedFile({
  large: "75x75",
  medium: "50x50",
  small: "25x25",
  contentType: /\Aimage\/.*\Z/ // or contentType: ["image/jpeg", "image/png"]
})
```

Furthermore, pushing this a bit further that library can define storage and strategy how to work with it:
```
FileUploader.storage = {
  storage: "s3",
  accessKeyId: 123,
  secretAccessKey: 456
}
```
